### PR TITLE
claims: Filter out Multus Default networks

### PR DESCRIPTION
**Pull request by oshoval**
**What this PR does / why we need it**:
We don't need to create claims for them,
As done already on the webhook as well.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

**Release note**:

```release-note

```

### CODE REVIEW

**Review by maiqueb — APPROVED** [review]

**Comment by maiqueb** [comment]
maybe add a unit test if it isn't too complex  ?

**Comment by oshoval** [comment]
> maybe add a unit test if it isn't too complex ?

checking

**Comment by oshoval** [comment]
btw we might need to also filter absent interfaces (not on this PR)

**Comment by oshoval** [comment]
Added a unit test, the question is if it is fine to use illegal config (can't be both pod and multus default at the same time)
the unit tests doesnt check this specifically, so it pass

File: pkg/vminetworkscontroller/vmi_controller_test.go [hunk 1]
```diff
@@ -330,6 +330,15 @@ func dummyVMISpec(nadName string) virtv1.VirtualMachineInstanceSpec {
...
 			},
+			{
+				Name: "default_multus",
+				NetworkSource: virtv1.NetworkSource{
+					Multus: &virtv1.MultusNetwork{
+						NetworkName: nadName,
+						Default:     true,
+					},
+				},
+			},
```
Comment (maiqueb) [inline pkg/vminetworkscontroller/vmi_controller_test.go:341]: this is a wrong VMI specification.

I don't understand what it means from the kubevirt perspective, but, the right configuration is to omit the pod network.

Please add another method `dummyVMISpecWithMultusDefaultNetwork` and use this one in a new test.

Comment (oshoval) [reply pkg/vminetworkscontroller/vmi_controller_test.go:341]: i know it is wrong i raised it here https://github.com/kubevirt/ipam-extensions/pull/46#issuecomment-2222586167 fwiw, wondered if it is fine to use it for this sake of testing

anyhow will do as you suggested

**Comment by oshoval** [comment]
updated, hope it is good enough (imo it is)

File: pkg/vminetworkscontroller/vmi_controller_test.go [hunk 2]
```diff
@@ -299,27 +306,29 @@ var _ = Describe("vmi IPAM controller", Serial, func() {
 	)
 })
 
-func dummyVM(nadName string) *virtv1.VirtualMachine {
+type VMISpecFunc func(nadName string) virtv1.VirtualMachineInstanceSpec
+
+func dummyVM(nadName string, specFunc VMISpecFunc) *virtv1.VirtualMachine {
```
Comment (maiqueb) [inline pkg/vminetworkscontroller/vmi_controller_test.go:311]: so you're giving it the nad name, and a function whose input is the nad name as parameters ...

why not simply give it the calculated spec, and pass it from the outside ? IMO it's less convoluted.

Comment (oshoval) [reply pkg/vminetworkscontroller/vmi_controller_test.go]: right done

**Comment by oshoval** [comment]
rebased

**Review by maiqueb — APPROVED** [review]
I'll leave to @qinqon to stamp the approval.

**Review by qinqon — APPROVED** [review]
/lgtm
/approve

**Pull request merged**